molecular/gfx/functions/Demultiplexer.h

```diff
@@ -0,0 +1,30 @@
+#ifndef MOLECULAR_GFX_DEMULTIPLEXER_H
+#define MOLECULAR_GFX_DEMULTIPLEXER_H
+
+#include <molecular/gfx/RenderFunction.h>
+
+namespace molecular
+{
+namespace gfx
+{
+
+/**
+ * @brief The Demultiplexer class is a utility class for cases where more than one output is required.
+ */
+class Demultiplexer : public MultipleCalleeRenderFunction
```

Comment:
"Output" in molecular terms is usually something like `gl_position`. So what does "output" mean here exactly?

Also, what does this class do that `MultipleCalleeRenderFunction` doesn't?

Reply:
Outputs here mean children. I'll update the docs to reflect that. This class allows for nodes which have only one child to be extended to have multiple children.

> Also, what does this class do that `MultipleCalleeRenderFunction` doesn't?

It actually calls all of its children. `SingleCalleeRenderFunction` works in the same way (leaves the call to the implementing classes) so I've decided to make `MultipleCalleeRenderFunction` to preserve consistency.

Reply:
I'd still change the name (why is it `Demultiplexer` and not `Multiplexer`?). Maybe just to `Group` or something similar.

Reply:
I've renamed it now.

`Multiplexer` selects one of possible signals and passes it on. What we are doing is more like a `Demultiplexer` because there is 1 input signal which then goes to one or more possible outputs, so I thought `Demultiplexer` was more appropriate :)